nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/FacetOperationsTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * (C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Ricardo Dias
+ */
+package org.nuxeo.ecm.automation.core.operations.document;
+
+import static org.junit.Assert.*;
+
+import com.sun.star.lang.IllegalArgumentException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.*;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import com.google.inject.Inject;
+import org.nuxeo.runtime.test.runner.LocalDeploy;
+
+/**
+ * @author rdias
+ */
+
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@Deploy("org.nuxeo.ecm.automation.core")
+@LocalDeploy("org.nuxeo.ecm.automation.core:OSGI-INF/add-facet-test-contrib.xml")
+public class FacetOperationsTest {
+
+    @Inject
+    CoreSession session;
+
+    @Inject
+    AutomationService service;
+
+    // MyNewFacet is declared in OSGI-INF/add-facet-test-contrib.xml
+    public static final String THE_FACET = "MyNewFacet";
+
+    protected DocumentModel folder;
+
+    protected DocumentModel docNoFacet;
+
+    protected DocumentModel docWithFacet;
+
+    @Before
+    public void initRepo() throws Exception {
+        session.removeChildren(session.getRootDocument().getRef());
```

Comment:
Do we need to clear the repository ? We're doing it at tear down already in `clearRepo`